### ISSUE TITLE
Add missing hieroglyph u16

### ISSUE THREAD

**Issue by nataliakeniganti** [Mercury13/unicodia#405]
![bandicam 2024-11-02 17-51-39-336](https://github.com/user-attachments/assets/c3d9bda3-cbc2-4623-a268-9916bc0fca0a)

**Comment by Mercury13** [comment]
The author of request PROMISED TO release his font. I don’t see it for now, so draw for yourself if you want.
1. Size — any but consistent.
2. Source — always specified, whether you drew it for yourself or took elsewhere.

**Comment by nataliakeniganti** [comment]
Working commit `UnicodiaHiero.ttf`
```
Loading error tofi fixing
```

**Comment by Mercury13** [comment]
Yes, I WAS working on transferring at least something into Unicode.
Not 4000, but 2900, from JSesh.
Stumbled on real problems: IDK how to auto-check correctness.
Need boolean operations with curves ELSEWHERE: FontForge constantly confuses direction.

**Comment by nataliakeniganti** [comment]
Correct newgardiner next version 2.10

**Comment by Mercury13** [comment]
> Correct new gardiner next version 2.10 Add label new unicode.

We call it “divide bear’s pelt before killing”. Or “cook a hare before catching” in English.
NewGardiner WAS my backup font when dozens of characters were wrong.
All work on Noto to bring U15.1 was MINE: I drew, some Egyptologist checked.

**Comment by Mercury13** [comment]
I periodically check if someone releases those 4 thousand. No fools :(

**Comment by nataliakeniganti** [comment]
unclosed not planned

**Comment by nataliakeniganti** [comment]
Working hieroglyphs 
```
13460
13461
13462
13463
13464
13465
13466
13467
13468
13469
1346A
1346B
1346C
1346D
1346E
1346F
13470
13471
13472
13473
13474
13475
13476
13477
13478
13479
1347A
1347B
1347C
1347D
1347E
1347F
13480
13481
13482
13483
13484
13485
13486
13487
13488
13489
1348A
1348B
1348C
1348D
1348E
1348F
13490
13491
13492
13493
13494
13495
13496
13497
13498
13499
1349A
1349B
1349C
1349D
1349E
1349F
134A0
134A1
134A2
134A3
134A4
134A5
134A6
134A7
134A8
134A9
134AA
134AB
134AC
134AD
134AE
134AF
134B0
134B1
134B2
134B3
134B4
134B5
134B6
134B7
134B8
134B9
134BA
134BB
134BC
134BD
134BE
134BF
134C0
134C1
134C2
134C3
134C4
134C5
134C6
134C7
134C8
134C9
134CA
134CB
134CC
134CD
134CE
134CF
134D0
134D1
134D2
134D3
134D4
134D5
134D6
134D7
134D8
134D9
134DA
134DB
134DC
134DD
134DE
134DF
134E0
134E1
134E2
134E3
134E4
134E5
134E6
134E7
134E8
134E9
134EA
134EB
134EC
134ED
134EE
134EF
134F0
134F1
134F2
134F3
134F4
134F5
134F6
134F7
134F8
134F9
134FA
134FB
134FC
134FD
134FE
134FF
13500
13501
13502
13503
13504
13505
13506
13507
13508
13509
1350A
1350B
1350C
1350D
1350E
1350F
13510
13511
13512
13513
13514
13515
13516
13517
13518
13519
1351A
1351B
1351C
1351D
1351E
1351F
13520
13521
13522
13523
13524
13525
13526
13527
13528
13529
1352A
1352B
1352C
1352D
1352E
1352F
13530
13531
13532
13533
13534
13535
13536
13537
13538
13539
1353A
1353B
1353C
1353D
1353E
1353F
13540
13541
13542
13543
13544
13545
13546
13547
13548
13549
1354A
1354B
1354C
1354D
1354E
1354F
13550
13551
13552
13553
13554
13555
13556
13557
13558
13559
1355A
1355B
1355C
1355D
1355E
1355F

```

**Comment by Mercury13** [comment]
Calm down, lady! They are absent EVERYWHERE!

**Comment by nataliakeniganti** [comment]
start in november 12-december-2024

**Comment by Mercury13** [comment]
Calm down! It’s better to check: how to import e.g. JSesh.

**Comment by nataliakeniganti** [comment]
Unicode block is not done

**Comment by nataliakeniganti** [comment]
Reopen issue?

**Comment by nataliakeniganti** [comment]
Add invalid label

**Comment by Mercury13** [comment]
Natalia, I’m thinking what to do, but 2900 random characters (out of 4000) is not what we expect from Unicodia.
All Egyptian fonts have murky licensing rules.

**Comment by Mercury13** [comment]
Tried to auto-fix 100 SVGs.
14 of them the machinery is unable to fix, reasons are.
Curves touch in one point due to sloppy editing: 2
Curves form a negative angle, and the system is unable to fix it due to roundoff errors: 4
Curves touch in one point intentionally:
… eye: 3
…flat object + smth else: 3
Unclassified artifacts (unintended intersections, round-off errors, etc): 3
(one SVG falls into two categories simultaneously)
So from 3000 SVGs I’ll probably need 450 hand-edited.

**Comment by Mercury13** [comment]
Cache contains 63 files. It means: out of 100, my software helped with 59 and failed with 14.
I managed to remove unclassified artifacts in one SVG, leaving negative angle only.

**Comment by Mercury13** [comment]
More work, 11 left
Curves unintentionally touch in one point due to sloppy editing: 2
Curves form a negative angle, and the system is unable to fix it due to roundoff errors: 1
Curves intentionally touch in one point:
…eye: 3
…flat object + smth else: 3
Big self-intersection machinery unable to erase: 1
Intersection between two curves so small that Inkscape sees nothing, and FontForge’s UI glitches: 1

**Comment by Mercury13** [comment]
Again, ≈330 glyphs is still much

**Comment by Mercury13** [comment]
Ran script on the entire directory. Actually 226 hieroglyphs are left for hand-editing.

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/6fbb3bfc-e59e-4d95-ab42-522775764b91)
Just a few hand-edited out of 200. 1070 tofu.

**Comment by nataliakeniganti** [comment]
Replace wontfix label to new unicode.

**Comment by nataliakeniganti** [comment]
Add missing 1070 tofu glyphs

**Comment by Mercury13** [comment]
Where to find them?
I repeat, there’ll be 1070 tofu out of 3995.

**Comment by Mercury13** [comment]
Want to help? Have Inkscape skills?
Download Unikemet base and JSesh, and start modifying.
If a character in Unikemet has a JSesh ID → IT’S ALREADY PRESENT, but you may take it as a base for other chars.
Work only on characters WITHOUT JSesh ID.

**Comment by Mercury13** [comment]
Not really 44. ONE THOUSAND AND SEVENTY. Really need workforce.

**Comment by nataliakeniganti** [comment]
Do 1026 tofu hieros

**Comment by Mercury13** [comment]
My log now shows 134 problem SVGs. The rest are remade by hand.

**Comment by Mercury13** [comment]
And 90 are remade by hand now.

**Comment by Mercury13** [comment]
My log now shows 134 problem SVGs. 90 are fixed by hand now.

**Comment by Mercury13** [comment]
The most common problems:
• curves marginally intersect, so Inkscape does not register intersection, and FontForge does
• curves intentionally touch
• curves form a negative angle, so Inkscape sees intersection, but my script fails to fix
• misc. glitches, probably by FontForge’s Simplify: e.g. saw two straights turned to wild curves

**Comment by nataliakeniganti** [comment]
Fix bad hieros
```
13A4B
13A6D
13A9A
13B56
13B6D
13B94
13C20
13C52
13C8A
13C8B
13CBB
13CC1
13D10
13D17
13D38
13DA5
13DA6
```

**Comment by nataliakeniganti** [comment]
Fix hand drawn 1070 tofu glyphs

**Comment by nataliakeniganti** [comment]
Add missing 1070 tofu glyphs

**Comment by Mercury13** [comment]
At least sucked all data I have.

**Comment by Mercury13** [comment]
Four nights of hard work. JSesh sucked, no major tech problems, but only partly checked.
Hieroglyphica stores data in local FireBird DB, ad IDK how to even check it. I suspect hieroglyphs are in bitmap format (Hieroglyphica is from Novosibirsk, and Delphi was highly popular in xUSSR, and FireBird is theirs going open-source).
I don’t even know who performs centerline bitmap tracing (rather than traditional outline).
The first glyph, 13460=A1F, even has no links to external bases.
Maybe I’ll draw seven hieros above the fold and switch to Tulu?

**Comment by Mercury13** [comment]
I’ll make seven hieros “above the fold” (1 found at JSesh, 4 done, 3 to go), and stop.
![image](https://github.com/user-attachments/assets/454ad789-ca6a-4ba3-8fbc-27fdb92393b5)

**Comment by Mercury13** [comment]
@nataliakeniganti
I use a button for selected item in table, mostly for the lack of anything better.
![image](https://github.com/user-attachments/assets/7dafa24d-0738-44ad-a64e-cda31824f960)
Please make a screenshot of table…
![image](https://github.com/user-attachments/assets/7c0f424b-378d-4f88-bf64-53741ce64686)
Probably that’s the cause of #414, but you refuse to work together.

**Comment by nataliakeniganti** [comment]
Add more 1060 hieros

**Comment by Mercury13** [comment]
Sorry, NO. I’ll do ≈10/week, but PLEASE READ what YOU can do for me.

**Comment by nataliakeniganti** [comment]
Calm down lady wait the 1060 hieros

**Comment by Mercury13** [comment]
@nataliakeniganti Nata, only 20 tofu remain. Wait for release.
